Locale/RequestDetector.php

```diff
@@ -47,12 +47,25 @@ public function __construct(ContainerInterface $container, $defaultLocale)
      */
     public function getLocale()
     {
-        if ($this->container->isScopeActive('request')) {
-            if ($request = $this->container->get('request', ContainerInterface::NULL_ON_INVALID_REFERENCE)) {
-                return $request->getLocale();
-            }
+        if ($request = $this->getRequest()) {
+            return $request->getLocale();
         }
 
         return $this->defaultLocale;
     }
+
+    /**
+     * @return Request|null
+     */
+    private function getRequest()
```

Comment:
This can be `getRequest` here. `handleRequest` on controller is different because it checks if passed request is null and create it if not.